### PR TITLE
 Add fetchSignInMethodsForEmail to auth-next

### PR DESCRIPTION
**Pull request by avolkovi**

### CODE REVIEW

File: packages-exp/auth-exp/src/core/strategies/email.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  // createAuthUri returns an error if continue URI is not http or https.
+  // For environments like Cordova, Chrome extensions, native frameworks, file
+  // systems, etc, use http://localhost as continue URL.
+  const continueUri = isHttpOrHttps() ? getCurrentUrl() : 'http://localhost';
+  const request: CreateAuthUriRequest = {
+    identifier: email,
+    continueUri
+  };
+
+  const response = await createAuthUri(auth, request);
```
Comment (samhorlbeck) [inline packages-exp/auth-exp/src/core/strategies/email.ts:38]: You can just destructure this here
`const {signInMethods} = await ...`

File: packages-exp/auth-exp/src/core/util/location.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+export function getCurrentUrl(): string {
+  return window?.location?.href || self?.location?.href || '';
```
Comment (samhorlbeck) [inline packages-exp/auth-exp/src/core/util/location.ts:19]: Isn't self just a direct alias to window?

Comment (avolkovi) [reply packages-exp/auth-exp/src/core/util/location.ts:19]: apparently window doesn't exist in workers but self does, this should probably just use the self version though

Comment (avolkovi) [reply packages-exp/auth-exp/src/core/util/location.ts:19]: https://developer.mozilla.org/en-US/docs/Web/API/Window/self

File: packages-exp/auth-exp/src/core/strategies/email.ts (hunk 1 above)
Comment (avolkovi) [reply packages-exp/auth-exp/src/core/strategies/email.ts]: done

**Pull request merged**